src/beam.ts

```diff
@@ -675,19 +732,19 @@ export class Beam extends Element {
   }
 
   // Get the x coordinates for the beam lines of specific `duration`
-  getBeamLines(duration) {
+  getBeamLines(duration: string): BeamInfo[] {
     const tick_of_duration = Flow.durationToTicks(duration);
-    const beam_lines = [];
+    const beam_lines: BeamInfo[] = [];
```

Comment:
As above, no need to define exported interface.

Here you can define a local type:

```typescript
type BeamInfo = { start: number, end?: number };
const beam_lines: BeamInfo[] = [];
...
```

Reply:
Global not exported is also fine?

Reply:
done local